packages/web3-eth-contract/src/utils.ts

```diff
@@ -31,6 +31,32 @@ import { isNullish, mergeDeep, toHex } from 'web3-utils';
 import { encodeMethodABI } from './encoding.js';
 import { ContractOptions, Web3ContractContext } from './types.js';
 
+const dataInputEncodeMethodHelper = (
+	txParams: TransactionCall,
+	abi: AbiFunctionFragment,
+	params: unknown[],
+): TransactionCall => {
+	let tx = txParams;
+	if (!isNullish(tx.input)) {
+		tx = {
+			...txParams,
+			input: encodeMethodABI(abi, params, txParams.input as HexString),
+		};
+	} else if (!isNullish(tx.data)) {
+		tx = {
+			...txParams,
+			data: encodeMethodABI(abi, params, txParams.data as HexString),
+		};
+	} else {
+		// default to using input
+		tx = {
+			...txParams,
+			input: encodeMethodABI(abi, params, txParams.input as HexString),
+		};
+	}
```

Comment:
What inside `else` is exactly what is inside the first `if`. Is this the correct logic? If yes, then it is better to merge them inside one block :smile: